### ISSUE TITLE
syntactic sugar request: Jacobian(Expression, Variables)

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#8287]
Related to https://github.com/robotlocomotion/drake/pull/8285

Currently we have `Expression::derivative(Variable)` and `Jacobian(VectorX<Expression>, Variables)`.  I'd like to request that we provide a derivative/Jacobian method (either name is ok with me) that takes the derivative of a Scalar w/ respect to a vector.  It would clean up the code considerably in both C++ and python.

In c++, it would change the second line of:
```
  const Expression V = x.dot(x);
  const Expression Vdot = (Jacobian(Vector1<Expression>(V), xvar) *
                         derivatives->CopyToVector())[0];
```
to
```
  const Expression Vdot = V.Jacobian(xvar).dot(derivatives->CopyToVector());
```

and in python, it woudl change the second line of:
```
V = x.dot(x)
Vdot = Jacobian([V], x).dot(dynamics(x))[0]
```
to 
```
Vdot = V.Jacobian(x).dot(dynamics(x))
```

In other words, it would get rid of both the cast to Vector1<Expression> and the cast back from the result (currently a vector) to the scalartype.

One subtlety - I think the return value of a jacobian of a scalar with respect to a vector of variables should be a rowvector.  Haven't actually checked that c++ eigen will do .dot() on that and know to return a scalar.
